Kudu.Services/Diagnostics/ProcessController.cs

```diff
@@ -241,90 +233,41 @@ public HttpResponseMessage MiniDump(int id, int dumpType = 0, string format = nu
         {
             using (_tracer.Step("ProcessController.StartProfileAsync"))
             {
+                // check if the process Ids exists in the sandbox. If it doesn't, this method returns a 404 and we are done.
                 var process = GetProcessById(id);
 
-                // Check if the profiling is already running for the given process. If it does, then just return with 200.
-                if (_profilingList.ContainsKey(process.Id))
-                {
-                    return Request.CreateResponse(HttpStatusCode.OK);
-                }
-
-                // Profiling service supports up to 255 concurrent profiling sessions.
-                if (_profilingList.Count >= 255)
-                {
-                    _tracer.TraceError("Too many profiling sessions are currently running.");
-                    return Request.CreateErrorResponse(HttpStatusCode.ServiceUnavailable, "Too many profiling sessions are currently running. Please try again later.");
-                }
-
-                int profilingSessionId = GetNextProfilingSessionId();
-
-                string processName = System.Environment.ExpandEnvironmentVariables("%SystemDrive%\\msvsmon\\profiler\\VSStandardCollector.Dev14.exe");
-                string arguments = System.Environment.ExpandEnvironmentVariables(string.Format("start {0} /attach:{1} /loadAgent:4EA90761-2248-496C-B854-3C0399A591A4;DiagnosticsHub.CpuAgent.dll  /scratchLocation:%LOCAL_EXPANDED%\\Temp", profilingSessionId, id));
+                var result = await ProfileManager.StartProfileAsync(process.Id, _tracer);
```

Comment:
leaving it as is per our conversation. async methods can't return out arguments.